### ISSUE TITLE
Add a message saying a bps patch failed if its CRC fails while applying softpatching

### ISSUE THREAD

**Issue by i30817** [libretro/RetroArch#13474]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

The title. I found that a bps patch being applied to the wrong ROM fails silently while attempting to help someone in the mesen issue report page saying that 'softpatching wasn't working'. I had it in my head that since one of the games i tried was ips and that worked, and a bps patch i was trying wasn't working (the reason why i looked for the bug) that 'bps didn't work in the mesen core', while the actual problem was that 'bps is failing silently with the wrong rom to softpatch'. @Sanaki sorted this misinformation out, but i kind of think that other people might have fallen into this trap.

https://github.com/libretro/Mesen/issues/11

### Expected behavior

BPS patches that fail to apply because of CRC mismatch popup a warning.

### Actual behavior

Nothing happens.

**Comment by Sanaki** [comment]
Seems reasonable to me. Of note, this applies to ups patches as well.

**Comment by i30817** [comment]
Does ips at least have a size check or nothing at all?

**Comment by Sanaki** [comment]
Nah, ips is as raw as it gets. "Throw changes at wall and hope it works." It doesn't validate checksum, size, or anything else.

**Comment by i30817** [comment]
I was hoping it had at least a implied size check, if not a explicit one. Oh well.
